From f6760169ccf0656555b0e2ee341f748d06f79352 Mon Sep 17 00:00:00 2001
From: Roy Pledge <roy.pledge@nxp.com>
Date: Fri, 22 Jun 2018 15:47:07 -0400
Subject: [PATCH 573/767] staging/fsl_qbman: Consume all frames in DQRR during
 init

The qm_dqrr_cdc_consume_n() function takes a bitmask, not an index.

Signed-off-by: Roy Pledge <roy.pledge@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl_qbman/qman_low.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_qbman/qman_low.h b/drivers/staging/fsl_qbman/qman_low.h
index e43f9ea0937f..4cd7fc558f1d 100644
--- a/drivers/staging/fsl_qbman/qman_low.h
+++ b/drivers/staging/fsl_qbman/qman_low.h
@@ -658,7 +658,7 @@ static inline int qm_dqrr_init(struct qm_portal *portal,
 			qm_dqrr_cce_consume(portal, dqrr->fill);
 			break;
 		case qm_dqrr_cdc:
-			qm_dqrr_cdc_consume_n(portal, (QM_DQRR_SIZE - 1));
+			qm_dqrr_cdc_consume_n(portal, (1<<QM_DQRR_SIZE) - 1);
 			break;
 		default:
 			DPA_ASSERT(0);
-- 
2.17.0

